src/services/completions.ts

```diff
@@ -953,6 +953,7 @@ namespace ts.Completions {
                 || tryGetConstructorCompletion()
                 || tryGetClassLikeCompletionSymbols()
                 || tryGetJsxCompletionSymbols()
+                || JSFileType()
                 || (getGlobalCompletions(), GlobalsSearch.Success);

```

Comment:
Don't think you want to add here. Instead you want to set the filter in `getGlobalCompletions` by checking  `isSourceFileJS(sourceFile)`